Separating Scala Actors from the Scala Library.

vjovanov opened the pull request:
Scala actors are now in scala-actors.jar. Changes that were done are:
- Fixed partest to include actors library for various test usages
- Created the entry for the new jar in build.xml
- Added maven entries for scala actors

Review by: @jsuereth

jsuereth commented:
So, +10 on modularizing out actors.   Quick question:  Are you creating a `scala-actors-docs.jar` with the scaladoc?  I didn't catch it in the diff.

Commit:
Separating Scala Actors from the Scala Library.

Scala actors are now in scala-actors.jar. Changes that were done are:
 - Fixed partest to include actors library for various test usages
 - Created the entry for the new jar in build.xml
 - Added maven entries for scala actors

Review by: @jsuereth

vjovanov commented:
Great catch, I missed it and it is fixed now. Currently we are copying the scala-library-doc.jar to be the same as scala-swing-doc.jar so I made the same hack for actors. Not sure should if this should be changed.

jsuereth commented:
It should, the issue is build time.  Scaladoc adds a lot, and this is "good
enough" for now....
On Mar 19, 2012 12:18 PM, "Vojin Jovanovic" <
reply@reply.github.com>
wrote:

> Great catch, I missed it and it is fixed now. Currently we are copying the
> scala-library-doc.jar to be the same as scala-swing-doc.jar so I made the
> same hack for actors. Not sure should if this should be changed.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/scala/scala/pull/288#issuecomment-4576877

soc commented on src/build/maven/scala-actors-pom.xml:
```diff
@@ -0,0 +1,62 @@
+      </license>
+   </licenses>
+   <scm>
+      <connection>scm:svn:http://lampsvn.epfl.ch/svn-repos/scala/scala/trunk</connection>
```
Should probably point to https://github.com/scala/scala.

vjovanov replied:
I will fix it in a separate pull request. There are multiple places where this fix needs to be applied. I was keeping it consistent and separating concerns.

soc commented on src/build/maven/scala-actors-pom.xml:
```diff
@@ -0,0 +1,62 @@
+   </scm>
+   <issueManagement>
+      <system>trac</system>
+      <url>http://lampsvn.epfl.ch/trac/scala
```
Should probably point to https://issues.scala-lang.org.

jsuereth replied:
Good call.  Thanks.

vjovanov replied:
I know. I did not change this as it needs to be changed in multiple places. I will make a separate pull request for that.
